### PR TITLE
BYOC

### PR DESCRIPTION
**Pull request by ElasticBottle**

### CODE REVIEW

File: src/components/PaperCheckout.tsx [hunk 1]
```diff
@@ -67,6 +75,9 @@ export const PaperCheckout: React.FC<PaperCheckoutProps> = ({
   recipientWalletAddress,
   emailAddress,
   quantity,
+  eligibilityMethod,
+  mintMethod,
+  signatureArgs,
```
Comment (edwardysun) [inline src/components/PaperCheckout.tsx:80]: Do we not need args for both eligibilityMethod and mintMethod?

Comment (ElasticBottle) [reply src/components/PaperCheckout.tsx:80]: What do you mean by need args for both mint and eligibility?

File: src/components/PayWithCard.tsx [hunk 2]
```diff
@@ -128,7 +155,9 @@ export const PayWithCard: React.FC<PayWithCardProps> = ({
           postMessageToIframe(payWithCardIframe, data.eventType, data);
           break;
 
-        case 'review':
+        case 'openReviewPaymentPopupWindow':
```
[the remark this answers is not in the document]
Comment (ElasticBottle) [reply src/components/PayWithCard.tsx:158]: Oh shoots, you're right, this is fine, but we need to keep emitting review in `paper-web`, will update

File: src/components/PayWithCard.tsx [hunk 3]
```diff
@@ -153,33 +177,47 @@ export const PayWithCard: React.FC<PayWithCardProps> = ({
...
-        'metadata',
-        encodeURIComponent(JSON.stringify(metadata)),
+        'mintMethod',
+        Buffer.from(mintMethodStringified, 'utf-8').toString('base64'),
+      );
+    }
+    if (eligibilityMethod) {
+      payWithCardUrl.searchParams.append(
+        'eligibilityMethod',
+        Buffer.from(eligibilityMethodStringified, 'utf-8').toString('base64'),
```
[the remark this answers is not in the document]
Comment (ElasticBottle) [reply src/components/PayWithCard.tsx:213]: Appending to search params automatically encodes the value as Uri component, I'll double check again just to be sure

File: src/interfaces/CustomContract.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
+export enum Currency {
+  MATIC,
+  ETH,
+  USDC,
+  'USDC.e',
```
[the remark this answers is not in the document]
Comment (ElasticBottle) [reply src/interfaces/CustomContract.ts:5]: Oh. I'm actl just using enums to get string type hints, this is actl such a roundabout way 🤦‍♀️

Do you prefer enums or string type hints? Also "USDe" seems better. I'll go change it. Was lazy to swap bc Edward set it up for Heretic so there's probably some consideration there that I'll have to think abt

File: src/interfaces/CustomContract.ts [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+  SOL,
+  AVAX,
+}
+export enum gasOptions {
+  'low',
+  'normal',
+  'high',
+}
```
[the remark this answers is not in the document]
Comment (ElasticBottle) [reply src/interfaces/CustomContract.ts:13]: Ya, this really should be just a string type hint

File: src/interfaces/CustomContract.ts [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+  MATIC,
+  ETH,
+  USDC,
+  'USDC.e',
+  SOL,
+  AVAX,
+}
+export enum gasOptions {
+  'low',
+  'normal',
```
[the remark this answers is not in the document]
Comment (ElasticBottle) [reply src/interfaces/CustomContract.ts:11]: 👍

File: src/components/PayWithCard.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (ElasticBottle) [reply src/components/PayWithCard.tsx:213]: image

for ref

**Pull request merged**